src/replacers/vars.js

```diff
@@ -57,11 +58,22 @@ function get(name, varsArr) {
   if (!Array.isArray(varsArr)) {
     throw new Error('You should pass vars array to vars.get()');
   }
+
+  const rootVar = name.match(/[^\.\[]*/)[0];
+
   // todo: use for.. of after https://github.com/facebook/react-native/issues/4676
   for (let i = 0; i < varsArr.length; i++) {
     let vars = varsArr[i];
-    if (vars && vars[name] !== undefined) {
-      return vars[name];
+    if (vars && vars[rootVar] !== undefined) {
+      if (rootVar.length ===  name.length) {
```

Comment:
I suggest to move this check out of `for`, give it name and compare it directly instead of length, e.g.
```js
const isSimpleVar = rootVar === name;
```

Also we can avoid nested `if` to make code cleaner
```js
  for (let i = 0; i < varsArr.length; i++) {
    let vars = varsArr[i];
    if (!vars || vars[rootVar] === undefined) {
       continue;
    }
    if (isSimpleVar) {
        return vars[name];
    } 
    try {
          return resolvePath({[rootVar]: vars[rootVar]}, name);
        } catch (error) {
          return undefined;
        }
     }
 }